accessibility/README.md

```diff
@@ -0,0 +1,495 @@
+* Ensure that the focus trap is cleared when an overlay is closed
+* Ensure that the focus trap is updated when the overlay state changes
+  * Example: if you have a multi-tiered navigation, ensure that the focus trap updates as the user navigates between tiers
+  * Example: if you have a search overlay and the results update as you type, ensure that focus trap updates as the content changes
+* Focus trapping can be achieved in Canvas projects with the `focusTrap` and `previousElement` helpers in the `accessibility.js` store - see [Canvas Gitbook - accessibility.js](https://we-make-websites.gitbook.io/canvas/features/vuex-stores/accessibility.js)
+* Focus trapping can be achieved in Frame 3 projects by using the `@shopify/theme-a11y` package - see [@shopify/theme-ally docs](https://github.com/Shopify/theme-scripts/blob/master/packages/theme-a11y/README.md)
+
+#### Why
+
+* Using a focus trap prevents users from tabbing to elements outside of the overlay element, preventing an issue where the user might not know which element they are focussed on
+
+[ꜛ Back to TOC](#table-of-contents)
+
+## Accessible Text
+
+* [Ensure that `input` elements have corresponding `label` elements](#ensure-that-input-elements-have-corresponding-label-elements)
+* [Ensure interactive elements have accessible text](#ensure-interactive-elements-have-accessible-text)
+* [Ensure correct usage of `aria-labelledby` and `aria-label`](#ensure-correct-usage-of-aria-labelledby-and-aria-label)
+
+### Ensure that `input` elements have corresponding `label` elements
+
+#### Don't
+
+```html
+<input
+  id="newsletter-email"
+  name="email" 
+  type="email"
+  placeholder="Enter your email"
+>
+```
+
+#### Do
+
+```html
+<label for="newsletter-email">
+  Enter your email
+</label>
+
+<input
+  id="newsletter-email"
+  name="email" 
+  type="email"
+  placeholder="Enter your email"
+>
+```
+
+* Ensuring that you have that an `input` has a corresponding `label` element helps to describe the purpose and can provide instructions for the input for users
+* A `label` should be used even when the input has a `placeholder` attribute.
+  * Assistive technology, such as screen-readers do not treat placeholder text as labels, so a `placeholder` should be used alongside a label, not instead of a label
+* In some cases, a site's design may show inputs without visible labels. Sometimes this may be because other elements on the page give the input some context - for example a search text input placed directly next to a "Search" button.
+  * If a input does not have a visible label, a `<label>` element should still be added for the benefit of screen readers
+  * A label can be hidden visually by adding the `visually-hidden` class to the `label`
+  ```html
+  <!-- Label will be available to assistive technology, but visually hidden -->
+  <label class="visually-hidden" for="search">
```

Comment:
Attributes, single lines, etc.

Not going to point out further cases, please update all instances.